### PR TITLE
Only reject if the corresponding rpcSate has a source

### PR DESCRIPTION
**Pull request by nibhar**
in conjunction with nimiq/rpc#22

Updates @nimiq/rpc to v0.3.0 and @nimiq/keyguard-client to v0.6.0.

requires nimiq/keyguard-next#333

### CODE REVIEW

File: src/lib/RpcApi.ts [hunk 1]
```diff
@@ -267,7 +267,9 @@ export default class RpcApi {
                 }
 
                 if (error.message === 'Request aborted') {
-                    if (window.opener) {
+                    const rpcState = RpcState.fromJSON(state.rpcState);
```
Comment (sisou) [inline src/lib/RpcApi.ts]: The rpcState is also available as `this._staticStore.rpcState`, because the state was already recovered in line 259.

Comment (nibhar) [reply src/lib/RpcApi.ts]: You are right. Fixed it.

File: src/lib/RpcApi.ts [hunk 2]
```diff
@@ -267,7 +267,13 @@ export default class RpcApi {
                 }
 
                 if (error.message === 'Request aborted') {
-                    if (window.opener) {
+                    /*
+                     * In case the window is an iFrame and the recovered state is the one with which the
+                     * iFrame was initialized (has a source), then reject it. The iFrame will be closed as a result.
+                     * If not there was another history entry in between, where a history.back() will navigate to,
+                     * not closing the iFrame in the process
```
Comment (danimoh) [inline src/lib/RpcApi.ts]: iframe or popup?

Comment (sisou) [reply src/lib/RpcApi.ts]: I also think you mean Popups :D

Comment (nibhar) [reply src/lib/RpcApi.ts]: Oh yes, I absolutely mean popup...

**Pull request merged**